lib/hq/graphql/filter_operations.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require "hq/graphql/enum/filter_operation"
+
+module HQ
+  module GraphQL
+    module FilterOperations
+      class Operation
+        attr_reader :name, :arel, :check_for_null, :sanitize
+
+        def initialize(name:, arel:, check_for_null: false, sanitize: nil)
+          @name           = name
+          @arel           = arel
+          @check_for_null = check_for_null
+          @sanitize       = sanitize
+        end
+
+        def sanitize_value(value)
+          sanitize ? sanitize.call(value) : value
+        end
+
+        def to_arel(table:, column_name:, value:)
+          sanitized_value = sanitize_value(value)
+
+          if arel.is_a?(Proc)
+            return arel.call(table: table, column_name: column_name, value: sanitized_value)
+          end
+
+          expression = table[column_name].send(arel, sanitized_value)
+
+          if check_for_null
+            expression = expression.or(table[column_name].eq(nil))
+          end
+
+          expression
+        end
+      end
+
+      OPERATIONS = [
+        EQUAL = Operation.new(name: "EQUAL", arel: :eq),
```

Comment:
ruby `arel` is a pretty good stuff

Reply:
A lot more arel in [here](https://github.com/OneHQ/has_reports/blob/master/app/generators/has_reports/sql_builder/visitors/to_arel.rb)! 😅😅😅